training_utils.py

```diff
@@ -377,4 +431,69 @@ def train_loop(
         print(f"Saved final model to {save_path}")
 
     print("Best Perplexities,", best_ppl)
-    return
\ No newline at end of file
+    return
+
+def reg_loop(
+    args,
+    model,
+    train_dataset,
+    device,
+    tokenizer=None,
+    regularizer=None
+):
+    # Quick hack to get tree regularization scores for a saved model
+    num_steps = 0
+    train_batch_size = args.batch_size
+    use_gold = args.use_gold
+    max_steps = args.max_train_steps
+
+    if tokenizer is not None:
+        train_data_collator = DataCollatorWithPadding(tokenizer=tokenizer)
+    else:
+        train_data_collator = collate.VarLengthCollate(tokenizer)
+
+    all_sci_scores = []
+
+    while True:
+        # print(train_dataset)
+        train_dataloader = DataLoader(
+            train_dataset,
+            sampler=RandomSampler(train_dataset),
+            batch_size=train_batch_size,
+            collate_fn=train_data_collator,
+        )
+        total_train_sz = len(train_dataset)
+        if num_steps > max_steps:
+            break
+        with torch.enable_grad(), tqdm(total=total_train_sz) as progress_bar:
+            for curr_batch_dict in train_dataloader:
+                num_steps += 1
+                if num_steps > max_steps:
+                    break
+                if type(model) != torch.nn.Module:
+                    model.model.train()
```

Comment:
nit: This can probably be checked outside of the loop to simplify.

Reply:
This is a holdout from Shikhar's code, some of the methods below access the Interface object, while others use the model itself.

Reply:
Ah gotcha, thanks for clarifying.